localstack-core/localstack/testing/scenario/provisioning.py

```diff
@@ -243,9 +243,10 @@ def provision(self, skip_deployment: Optional[bool] = False):
                 ]
 
                 for s3_bucket in s3_buckets:
+                    bucket_to_delete = s3_bucket  # Capture current value
```

Comment:
FYI: this line doesn't actually  do anything in regards to the root cause, it would still lead to the same issue

Reply:
Thanks for the suggestion. I've removed this line.